apis/Google.Cloud.DevTools.Common/Google.Cloud.DevTools.Common/Source/V1/SourceContext.File.cs

```
@@ -0,0 +1,104 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using System;
+using System.IO;
+using System.Security;
+
+namespace Google.Cloud.DevTools.Source.V1
+{
+    /// <summary>
+    /// Partial class to SourceContext proto.
+    /// The class here reads source-context.json file from app root.
+    /// </summary>
+    public sealed partial class SourceContext
+    {
+        private const string SourceContextFileName = "source-context.json";
+        private static Lazy<string> s_filePath = new Lazy<string>(GetFilePath);
+        private static Lazy<SourceContext> s_sourceContext = new Lazy<SourceContext>(OpenParseFile);
+
+        /// <summary>
+        /// Gets the custom log label of Stackdriver Logging entry for Git commit id.
+        /// </summary>
+        public const string GitRevisionIdLogLabel = "git_revision_id";
+
+        /// <summary>
+        /// Gets the <seealso cref="SourceContext"/> for the application.
+        /// </summary>
+        public static SourceContext AppSourceContext => s_sourceContext.Value;
+
+        /// <summary>
+        /// Open the source context file and parse it with <seealso cref="SourceContext"/> proto.
+        /// </summary>
+        /// <returns>
+        /// A <seealso cref="SourceContext"/> object if the file is read and parsed successfully.
+        /// null is returned if there is error reading/parsing the file or the source context file is not found.
+        /// </returns>
+        private static SourceContext OpenParseFile()
+        {
+            string sourceContext = Read();
+            if (sourceContext == null)
+            {
+                return null;
+            }
+
+            try
+            {
+                return JsonParser.Default.Parse<SourceContext>(sourceContext);
+            }
+            catch (Exception ex) when (IsProtobufParserException(ex))
+            {
+                return null;
```

I understand the concern but this is a library that other may consume and it swallowing a lot of errors.  How will clients ever get this info if all they get back is null?

Log4Net is a client and IMO should handle this case and maybe log the failure.  If the exceptions are documented I think it should be okay.